network/wsNetwork_test.go

```diff
@@ -3757,3 +3759,106 @@ func TestWebsocketNetworkTelemetryTCP(t *testing.T) {
 	t.Log("closed detailsA", string(pcdA))
 	t.Log("closed detailsB", string(pcdB))
 }
+
+/*
+Exercises the updatePhonebookAddresses function, notably with different variations of valid relay and
+archival addresses.
+*/
+func TestUpdatePhonebookAddresses(t *testing.T) {
+	var netA *WebsocketNetwork
+
+	rapid.Check(t, func(t1 *rapid.T) {
+		netA = makeTestWebsocketNode(t)
+		relayPeers := netA.GetPeers(PeersPhonebookRelays)
+		assert.Equal(t, 0, len(relayPeers))
+
+		archivePeers := netA.GetPeers(PeersPhonebookArchivers)
+		assert.Equal(t, 0, len(archivePeers))
+
+		domainGen := rapidgen.Domain()
+
+		// Generate between 0 and N examples - if no dups, should end up in phonebook
+		domainsGen := rapid.SliceOfN(domainGen, 0, 200)
+
+		relayDomains := domainsGen.Draw(t1, "relayDomains").([]string)
+		archiveDomains := domainsGen.Draw(t1, "archiveDomains").([]string)
```

Comment:
nit: I realize this is in test code, but if it became invalid the failure would be a panic. An assert failure is easier to debug:
```suggestion
		draw := func(name string) []string {
			domains := domainsGen.Draw(t1, name)
			assert.IsType(t, []string{}, domains)
			return domains.([]string)
		}
		relayDomains := drraw("relayDomains")
		archiveDomains := draw("archiveDomains")
```

Reply:
I actually had to debug this while writing newer tests - the errors were clean and easy to follow. In this case, I'd like to optimize for brevity given that it is test code. The exact line numbers were there and all.